common/djangoapps/student/management/commands/bulk_change_enrollment_csv.py

```diff
@@ -0,0 +1,111 @@
+"""
+Management command to change many user enrollments in many courses using
+csv file.
+"""
+import csv
+import logging
+from os import path
+
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+
+from student.models import CourseEnrollment, CourseEnrollmentAttribute, User
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class Command(BaseCommand):
+    """
+        Management command to change many user enrollments in many
+        courses using the csv file
+    """
+
+    help = """
+        Change the enrollment status of all the users specified in
+        the csv file in the specified course to specified course
+        mode.
+        Could be used to update effected users by order
+        placement issues. If large number of students are effected
+        in different courses.
+        Similar to bulk_change_enrollment but uses the csv file
+        input format and can enroll students in multiple courses.
+
+        Example:
+            $ ... bulk_change_enrollment_csv csv_file_path
+        """
+
+    def add_arguments(self, parser):
+        """ Add argument to the command parser. """
+        parser.add_argument(
+            '--csv_file_path',
+            required=True,
+            help='Csv file path'
+        )
+
+    def handle(self, *args, **options):
+        """ Main handler for the command."""
+        file_path = options['csv_file_path']
+
+        if not path.isfile(file_path):
+            raise CommandError("File not found.")
+
+        with open(file_path, 'rb') as file:
+            file_reader = csv.DictReader(file)
+            for row in file_reader:
+                try:
+                    course_key = CourseKey.from_string(row['course_id'])
+                except InvalidKeyError:
+                    raise CommandError('Invalid or non-existant course id {}'.format(row['course_id']))
+
+                try:
+                    user = User.objects.get(username=row['user'])
+                except:
+                    raise CommandError('Invalid or non-existant user {}'.format(row['user']))
+
+                try:
+                    course_enrollment = CourseEnrollment.get_enrollment(user, course_key)
+                    #  If student is not enrolled in course enroll the student in free mode
+                    if not course_enrollment:
+                        # try to create a enroll user in default course enrollment mode in case of
+                        # professional it will break because of no default course mode.
+                        try:
+                            course_enrollment = CourseEnrollment.get_or_create_enrollment(user=user,
```

Comment:
we already checked for existent enrollment, here I guess you can only use `create`

Reply:
No existing method for create. using get_or_create_enrollment as it enrolls user in free mode.